lib/resources/security_policy.rb

```diff
@@ -142,10 +150,17 @@ def extract_value(val)
       if val =~ /^\d+$/
         val.to_i
       # special handling for SID array
-      elsif val =~ /^\*\S/
-        val.split(',').map { |v|
-          v.sub('*S', 'S')
-        }
+      elsif val =~ /[,]{0,1}\*\S/
+        if @translate_sid
+          val.split(',').map { |v|
+            temp = inspec.command("(New-Object System.Security.Principal.SecurityIdentifier(\"#{v.sub('*S', 'S')}\")).Translate( [System.Security.Principal.NTAccount]).Value").stdout.to_s.strip
```

Comment:
A better variable name here would be `policy_name` or something similar to that, rather than `temp`